api/app/use_cases/analysis/tree_cover/tree_cover_service.py

```diff
@@ -0,0 +1,45 @@
+from app.models.common.query import Filter, Query
+from app.models.land_change.tree_cover import TreeCoverAnalyticsIn
+
+
+class TreeCoverService:
+    def __init__(self, utils=None):
+        pass
+
+    def do(self, tree_cover_analytics: TreeCoverAnalyticsIn):
+        pass
+
+    def compute(self, tree_cover_analytics: TreeCoverAnalyticsIn):
+        """
+        Placeholder for the actual computation logic.
+        This method should implement the logic to compute tree cover analytics.
+        """
+        pass
+
+    def create_query_from_analytics(self, tree_cover_analytics: TreeCoverAnalyticsIn):
```

Comment:
@gtempus something to discuss in huddle tomorrow, here's an initial stab for the GFW services of abstracting the query from the actual Data API call. Basically, pass around a Query data class that contains the info needed to execute a query, whether it's on Data API or from a DuckDB table. And then we can have adapters for each of those that transform it into the appropriate SQL query for that repository.

Is that kind of getting at the ports and adapters idea? This way, we have just data we can pass around and unit test against. I think we'd also need to come up with models to describe datasets and fields in our codebase, since "tree_cover" might have a different name depending on the data source.